be/src/controlers/rest.router.ts

```diff
@@ -1,77 +1,121 @@
 import {Request, Response, Router} from 'express';
-import {vocabularyService} from "../services/vocabulary.service";
-import {Vocabulary} from "../models/dbo.models";
-import {Vocabulary as VocabularyDTO} from "../generated/models/Vocabulary";
+import {vocabularyService} from '../services/vocabulary.service';
+import {Vocabulary} from '../models/dbo.models';
+import {Vocabulary as VocabularyDTO} from '../generated/models/Vocabulary';
+import ListQueryModel from '../models/list-query.model';
 
 const router: Router = Router();
 
-function vocabDto2Dbo(dto: VocabularyDTO): Vocabulary {
-  const vocab: Vocabulary = {
-    _id: undefined,
-    created: undefined,
-    description: dto.description,
-    label: dto.label,
-    lastModified: undefined
-  }
-  return vocab
-}
-
-function vocabDbo2Dto(dbo: Vocabulary): VocabularyDTO {
-  return {
-    id: dbo._id.toHexString(),
-    label: dbo.label,
-    description: dbo.description,
-    created: dbo.created.toISOString(),
-    lastModified: dbo.lastModified.toISOString(),
-    entityCount: -1
-  };
-}
-
-router.get('/vocab', async (req: Request, res: Response) => {
-  res.json({statusCode:200});
-});
+router.get('/vocab', (req: Request, res: Response) => processVocab(req, res));
+
+router.get('/vocab/:id', (req: Request, res: Response) => processVocab(req, res, true));
```

Comment:
`GET /vocab` and `GET /vocab/:id` do **not** have the same response format.

The first is a list (with paging info, etc), the second is a single entity.

Reply:
Bugfix accomplished, please conduct a validation. 

Empty request on empty DB or with non existing ID return now:
![Screenshot from 2021-12-13 10-46-00](https://user-images.githubusercontent.com/8605616/145789520-9a403dc9-5ecf-4403-bab3-3a219dd91286.png)
![Screenshot from 2021-12-13 10-46-12](https://user-images.githubusercontent.com/8605616/145789532-955c8839-af7c-440e-8058-d7eba2b741e4.png)

Same with some DB entries and an a GET of a single vocab:
![Screenshot from 2021-12-13 10-46-35](https://user-images.githubusercontent.com/8605616/145789734-01edefe8-f537-432d-a970-2c99d02fed8b.png)
![Screenshot from 2021-12-13 10-46-46](https://user-images.githubusercontent.com/8605616/145789743-31c476f4-4f60-4033-bbe0-6ddbb65ac7a1.png)